### ISSUE TITLE
Add JSONP support to http response node

### ISSUE THREAD

**Issue by ghost** [node-red/node-red#102]
Hi
using node to replace some simple php. using localhost:1880 to make an ajax call gets stuck on a Cross Origin Domain error so trying to use JSONP instead.
## My query ajax call is this:

```
$.ajax({

    type : "GET",
    dataType : "jsonp",
    url : ajaxURL,
    data : {
        "firstname" : "Rudolph",
        "age" : 28,
        "livesin" : "Christmas Island",
        "favouritecolour" : "red"
    },
    success : function(data) {
        $.each( data, function( index, val ) {
        $('#jquerytest').append('<p> Name:' + val.firstname + ' , Age: ' + val.age +  ', Lives In: ' + val.livesin + ', Favourite Colour: ' + val.favouritecolour + '</p>');
        });
    }
});
```
## 

The flow basically returns what I sent to it - 
[{"id":"b7a04cb2.485fb","type":"mqtt-broker","broker":"my_special_broker","port":"1883","clientid":"me_here"},{"id":"1d2e0609.e2d1fa","type":"http in","name":"saveMongoData","url":"/saveMongoData","method":"get","x":90,"y":43,"z":"9051934e.6fae7","wires":[["8e4d59e8.71b2a8","19717e2d.e68e82","5727877e.a8d878"]]},{"id":"8e4d59e8.71b2a8","type":"http response","name":"","x":585,"y":156,"z":"9051934e.6fae7","wires":[]},{"id":"19717e2d.e68e82","type":"debug","name":"","active":true,"complete":false,"x":587,"y":89,"z":"9051934e.6fae7","wires":[]},{"id":"5727877e.a8d878","type":"mqtt out","name":"/red/out","topic":"/red/out/","broker":"b7a04cb2.485fb","x":638,"y":421,"z":"9051934e.6fae7","wires":[]}]
## 
## I get a Uncaught SyntaxError: Unexpected token :  in the browser which references saveMongoData:2 which contains this:

{
  "callback": "jQuery110206506511166226119_1386003865031",
  "firstname": "Rudolph",
  "age": "28",
  "livesin": "Christmas Island",
  "favouritecolour": "red",
  "_": "1386003865032"
}
## 

My question is whether JSONP is possible with NodeRed - or do I need to set up the CORS header detail before making the ajax call in my javascript - anyone got JSON or JSONP cross origin URL calls working using JSONP?

Thanks a bunch.
R

**Comment by shooftie** [comment]
Hi, sorry to stoke an old fire but I am in a similar situation. I think.

My situation is that I need the http response to be JSON and to be an array of objects.

I have therefore populated an array of objects in a function and assign this to `msg.payload`.

I then pass this into a `json` node, however this fails because `msg.payload` must be an object rather than an array.

So when I change the initial assignment to be an object of objects then I end up with that at the http response stage.

TLDR;

How do you return an array of objects as the http response when the payload can only be an object?

**Comment by knolleary** [comment]
@shooftie just pass the array to the HTTP Response node - it will encode it properly for you and set the appropriate content-type header to indicate it is json without you having to use the JSON node.

I'm not immediately clear why the JSON node rejects array payloads - but sure enough, there's a line of code explicitly doing that... will talk to @dceejay to check I'm not overlooking a reason.

**Comment by shooftie** [comment]
As simple as that... at this time on a Friday night you have made my weekend! Many thanks

**Comment by knolleary** [comment]
@shooftie and with 83a3642c0e526f5c34a5920b726f9bf3ab17316e in the next release, the JSON node will behave properly as well.

**Comment by shooftie** [comment]
I'm absolutely loving node-red at the moment. I've been through every method known to man and only node-red has made the process manageable. While I'm here can you suggest any good resources for tutorials/etc?

**Comment by calderonroberto** [comment]
@shooftie: there are tons of resources out there. You can always start with the documentation (http://nodered.org/docs/) and flows library (http://flows.nodered.org/). SenseTecnic also has a selection of tutorials (http://developers.sensetecnic.com/article/tag/node-red/). A couple of us are also working on a Node-RED lecture series to get people started; they are almost ready and we can let you know here when they are ready.

**Comment by shooftie** [comment]
Nice one...thanks for that. I've got to give this a rest for now though, I think I had a girlfriend around here somewhere.
